uptane/clients/primary.py

```diff
@@ -934,18 +927,69 @@ def _check_ecu_serial(self, ecu_serial):
-    through another interface and passed through the XMLRPC interface).
-
-    The Primary need not track ECU serials, so calling this doesn't result in a
-    verification of the ECU's signature on the ECU manifest. This information
-    is bundled together in a single vehicle report to the Director service.
-
-    Exceptions:
-      uptane.Spoofing     if ECU Serials within the manifest are inconsistent
-      uptane.UnknownECU   if the manifest is from an ECU that is not one of our
-                          Secondaries
-      uptane.Error        if the VIN for the report is not the same as our VIN
+    <Purpose>
+      Called by Secondaries (in the demo, this is via an XMLRPC interface, or
+      through another interface and passed through the XMLRPC interface).
+
+      The Primary need not track ECU keys, so calling this doesn't result in a
+      verification of the ECU's signature on the ECU manifest. This information
+      is bundled together in a single vehicle report to the Director service.
+
+    <Arguments>
+      vin
+          See class docstring above. The VIN of a Secondary in this vehicle
+          submitting an ECU Manifest is expected to be the same as the VIN for
+          this Primary. (In deployments where a Primary is shared -- for
+          example, a dealer device connected directly to a vehicle for manual
+          updates/modifications -- some code would have to be changed in a few
+          modules to remove this assumption.)
+
+      ecu_serial
+          The ECU Serial of the Secondary submitting the ECU Manifest. This
+          should match the ECU Serial listed in the signed manifest itself.
+
+      nonce
+          A (probably randomly generated) integer token produced by the
+          Secondary, which this Primary is expected to include in a request to
+          the Timeserver to produce a signed time that includes this token (and
+          others). When the Secondary receives the signed timeserver
+          attestation, if it sees this token in the signed contents of the
+          attestation, the Secondary can be reassured of the freshness of the
+          time attestation.
+
+      signed_ecu_manifest
+          The ECU Manifest a Secondary is submitting.
+
+      force_pydict (optional, default False)
+          When True, indicates that signed_ecu_manifest is a JSON-compatible
+          Python dictionary, the internal representation of an ECU Manifest,
+          conformant with uptane.formats.SIGNABLE_ECU_VERSION_MANIFEST_SCHEMA,
+          just as if tuf.conf.METADATA_FORMAT were set to json'. By default
+          (False), the expected format of signed_ecu_manifest is based on
+          tuf.conf.METADATA_FORMAT, so signed_ecu_manifest is expected to
+          conform to uptane.formats.DER_DATA_SCHEMA, which decodes to data
+          conformant with uptane.encoding.asn1_definitions.ECUVersionManifest
+
+
+    <Exceptions>
+
+      uptane.Spoofing
```

Comment:
I think this function can potentially raise a FormatError.  Also, why don't you validated all of the arguments (e.g., `vin`)?

Reply:
Thanks, good catch.